packages/@tinacms/graphql/src/database/datalayer.ts

```diff
@@ -132,14 +136,26 @@ const makeKeyForField = <T extends object>(
     ) {
       // TODO I think these dates are ISO 8601 so I don't think we need to convert to numbers
       const rawValue = data[field.name]
-      const resolvedValue = String(
-        field.type === 'datetime'
-          ? new Date(rawValue).getTime()
-          : field.type === 'string'
-          ? stringEscaper(rawValue as string | string[])
-          : rawValue
-      ).substring(0, maxStringLength)
-      valueParts.push(applyPadding(resolvedValue, field.pad))
```

Comment:
Note that this change is basically just making explicit what was already happening when an array of strings was cast to a string (ie ['foo','bar'] -> 'foo,bar')